### ISSUE TITLE
Support conversion to Array or contiguous Tensor

### ISSUE THREAD

**Issue by letmaik** [alex-weaver/TensorSharp#4]
I tried to convert Tensor objects to plain Array instances, however this seems nontrivial.

One way could be to add support for copying a Tensor and allowing it to become contiguous in the process (in case it's not already), then converting to Array is trivial by just copying the underlying storage buffer via `Storage.CopyFromStorage`.

Another way would be to read value-by-value with `GetElementAsFloat` and populate an Array instance, but this doesn't feel right and would probably be quite slow.

How hard would it be to add such a feature?

**Comment by alex-weaver** [comment]
Good point, Tensor->Array seems to be missing. The first approach I think is a reasonable way to achieve this. `GetElementAsFloat` is really only intended for convenience, you're right that it's performance iterating over an array will be less than ideal.

you can make sure you have a contiguous tensor with
```
using (var inputContig = Ops.AsContiguous(input))
{
   // use CopyFromStorage here
}
```

Using CopyFromStorage would need to make the same assumptions as `Tensor.FromArray` re. there being no platform-specific padding, but this is probably reasonable.

**Comment by letmaik** [comment]
Yes, that worked, thanks! I wasn't aware of AsContiguous.
